common/src/main/java/feast/common/interceptors/GrpcMessageInterceptor.java

```diff
@@ -59,22 +55,25 @@
     entryBuilder.setIdentity(identity);
 
     // Register forwarding call to intercept outgoing response and log to audit log
+    // As sendMessage(), close() and onMessage() may not be called in order (async)
+    // add a try log message call to each so that message logging would not
+    // depend on the calls to made in a specific order.
     call =
         new SimpleForwardingServerCall<ReqT, RespT>(call) {
           @Override
           public void sendMessage(RespT message) {
-            // 2. Track the response & Log entry to audit logger
+            // Track the response & Log entry to audit logger
             super.sendMessage(message);
```

Comment:
Should this `super.sendMessage(message);` not be the last call that is made?

Reply:
This was done to ensure that logging code, as much as possible stays out of the way of the rest of message (ie exceptions in logging code do not disrupt message handling.) Going to try as last call to see if it fixes the concurrency issue.

Reply:
From my testing, moving the location of statements does not resolve the issue.